src/CoreWCF.WebHttp/src/CoreWCF/OpenApi/Middleware/OpenApiDocumentProvider.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Options;
+
+namespace CoreWCF.OpenApi.Middleware
+{
+    internal sealed class OpenApiDocumentProvider
+    {
+        private readonly IOptionsMonitor<OpenApiOptions> _options;
+        private string _document;
+
+        public OpenApiDocumentProvider(
+            IOptionsMonitor<OpenApiOptions> options)
+        {
```

Comment:
You are using an IOptionsMonitor which is designed to allow runtime changes in the configuration. e.g if options is driven from a json config file, then you can be notified when that file has changed. The Document property builds the document once and doesn't rebuild on option changes. Switch to using IOptions instead and use the Value property instead of _options.CurrentValue.

Reply:
This has been addressed.